### PR TITLE
Add influxdb & smtp configurations

### PR DESCRIPTION
**Pull request by mhoppa**
## What does this PR (Pull Request) do?

This PR adds the ability to setup configurations for influxdb and smtp within the lab deployment Ansible playbook 

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: infrastructure/ansible/roles/traffic_ops/templates/cdn.conf.j2 [hunk 1]
```diff
@@ -45,6 +46,12 @@
       "email_from" : "{{ to_email_from }}",
       "no_account_found_msg" : "{{ to_account_missing_msg }}"
    },
+   "smtp" : {
+      "enabled" : {{ to_smtp_enabled | lower }},
```
Comment (jhg03a) [inline infrastructure/ansible/roles/traffic_ops/templates/cdn.conf.j2]: The `to_smtp_enabled` variable should be treated as a boolean (and enforced to run through a boolean filter) before running through `lower` to get `true || false`

File: infrastructure/ansible/roles/traffic_ops/templates/influxdb.conf.j2 [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#}
+{
+    "cache_stats_db_name": "{{ to_influxdb_cache_stats_db_name  }}",
+    "delivery_stats_db_name": "{{ to_influxdb_deliveryservice_stats_db_name }}",
+    "password": "{{ to_influxdb_password }}",
+    "user": "{{ to_influxdb_user }}",
+    "secure": {{ to_influxdb_secure | lower }}
```
Comment (jhg03a) [inline infrastructure/ansible/roles/traffic_ops/templates/influxdb.conf.j2]: The `to_influxdb_secure` variable should be treated as a boolean (and enforced to run through a boolean filter) before running through lower to get true || false

File: infrastructure/ansible/roles/traffic_ops/defaults/main.yml [hunk 3]
```diff
@@ -101,6 +102,10 @@ to_go_read_header_timeout: 120
 to_go_read_timeout: 120
 to_go_write_timeout: 120
 
+to_smtp_enabled: false
+to_smtp_username: ""
```
Comment (jhg03a) [inline infrastructure/ansible/roles/traffic_ops/defaults/main.yml:106]: Unless blank is meaningful, a default should exist.

File: infrastructure/ansible/roles/traffic_ops/defaults/main.yml [hunk 4]
```diff
@@ -101,6 +102,10 @@ to_go_read_header_timeout: 120
...
+to_smtp_password: ""
```
Comment (jhg03a) [inline infrastructure/ansible/roles/traffic_ops/defaults/main.yml:107]: Unless blank is meaningful, a randomly generated password should be used.

File: infrastructure/ansible/roles/traffic_ops/defaults/main.yml [hunk 5]
```diff
@@ -150,6 +155,13 @@ to_pi_health_thresh_load_avg: "25"
 to_pi_health_thresh_kbps: "1750000"
 to_pi_health_connect_timeout: "2000"
 
+# --- influxdb.conf
+to_influxdb_user: ""
+to_influxdb_password: ""
```
Comment (jhg03a) [inline infrastructure/ansible/roles/traffic_ops/defaults/main.yml]: Unless blank is meaningful, a randomly generated password should be used.

File: infrastructure/ansible/roles/traffic_ops/defaults/main.yml [hunk 6]
```diff
@@ -150,6 +155,13 @@ to_pi_health_thresh_load_avg: "25"
 to_pi_health_thresh_kbps: "1750000"
 to_pi_health_connect_timeout: "2000"
 
+# --- influxdb.conf
+to_influxdb_user: ""
```
Comment (jhg03a) [inline infrastructure/ansible/roles/traffic_ops/defaults/main.yml]: Unless blank is meaningful, a default should exist.

**Review by jhg03a — CHANGES_REQUESTED** [review]

File: infrastructure/ansible/roles/traffic_ops/defaults/main.yml (hunk 3 above)
Comment (mhoppa) [reply infrastructure/ansible/roles/traffic_ops/defaults/main.yml:106]: username and password can be blank for smtp as some smtp servers dont require it to authenticate I would rather leave these empty specially given the default for smtp_enabled is false.

**Review by jhg03a — APPROVED** [review]

**Pull request merged**